03 mongodb

Yakomoga777 opened the pull request:
Під час деплоя проєкту отримав помилку:

Jul 3 04:08:24 PM error commander@11.0.0: The engine "node" is incompatible with this module. Expected version ">=16". Got "14.17.0"
Jul 3 04:08:24 PM error Found incompatible module.
Jul 3 04:08:24 PM info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Jul 3 04:08:24 PM ==> Build failed 😞

Як я зрозумів, йому не подобалася моя версія commander. Я видалив цей пакет, так як він нам вже не потрібний.
Якщо я правильно розумію процес, то я маю Вам відправити пулреквест, Ви його приймете, і лише потім я зможу спробувати ще раз запустити деплой з master.

Yakomoga777 committed:
init project

Yakomoga777 committed:
removes commander@11

KhYulian commented:
> Під час деплоя проєкту отримав помилку:
> 
> Jul 3 04:08:24 PM error commander@11.0.0: The engine "node" is incompatible with this module. Expected version ">=16". Got "14.17.0" Jul 3 04:08:24 PM error Found incompatible module. Jul 3 04:08:24 PM info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command. Jul 3 04:08:24 PM ==> Build failed 😞
> 
> Як я зрозумів, йому не подобалася моя версія commander. Я видалив цей пакет, так як він нам вже не потрібний. Якщо я правильно розумію процес, то я маю Вам відправити пулреквест, Ви його приймете, і лише потім я зможу спробувати ще раз запустити деплой з master.

оновіть версію ноди на сервісі деплою. Деплой не є обов'язковим для виконання дз. Можете зараз зосередитись над самими дз

KhYulian commented on app.js:
```diff
@@ -1,18 +1,23 @@
+mongoose
+  .connect(DB_HOST)
+  .then(() => console.log("Database connect success"))
+  .catch((error) => console.log(error.message));
```
для чого підключатись до бд в app.js

KhYulian commented on server.js:
```diff
@@ -1,5 +1,19 @@
-  console.log("Server running. Use our API on port: 3000");
-});
+const DB_HOST =
+  "mongodb+srv://Yakomoga:vskmqjyth111111@cluster0.escdzh4.mongodb.net/db-contacts?retryWrites=true&w=majority";
```
використовуйте змінні серидовища для зберігання чутливої інформації

Yakomoga777 committed:
Підключає сервер до  mongoDB:

Створює приватні файли навколишнього оточення та підключає сервер до бази даних

Yakomoga777 committed:
update:
рефакторить додаток. Додає servises та conrollers

Yakomoga777 committed:
update:
Додає схеми валідації, додатковий роут, та функціїю заміни favorite

Yakomoga777 committed:
Update:
Видаляє contact.json

KhYulian reviewed (approved):
Чудова робота
